### ISSUE TITLE
TEST-19-DELEGATE is failing with `code=exited/status=1` on Ubuntu Xenial

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#7440]
It seems that `TEST-19-DELEGATE` fails to pass when the kernel doesn't support `cgroup2`.
```
test0.service: User lookup succeeded: uid=64623 gid=64623
Successfully mounted /tmp/systemd-private-ad376b24eb5e4f07a4c9fe7ae318eae3-test0.service-rrUwSb/tmp to /tmp
Successfully mounted /var/tmp/systemd-private-ad376b24eb5e4f07a4c9fe7ae318eae3-test0.service-hb6WJI/tmp to /var/tmp
test0.service: Executing: /usr/bin/test -w /sys/fs/cgroup/system.slice/test0.service/ -a -w /sys/fs/cgroup/system.slice/test0.service/cgroup.procs -a -w /sys/fs/cgroup/system.slice/test0.service/cgroup.subtree_control
Preset files don't specify rule for test0.service. Enabling.
audit: type=1131 audit(1511447403.817:26): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=test0 comm="systemd" exe="/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=test0 comm="systemd" exe="/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
test0.service: Child 429 belongs to test0.service
test0.service: Main process exited, code=exited, status=1/FAILURE
```

**Comment by evverx** [comment]
I think that the test shouldn't check whether `cgroup.subtree_control` is writable when `cgroup2` is not supported, because `cgroup.subtree_control` is a `cgroup2` thing.

**Comment by evverx** [comment]
Judging by `cgroup.subtree_control`, `UNIFIED_CGROUP_HIERARCHY=yes`, 
 and `/sys/fs/cgroup/system.slice/*`, the test is not supposed to be run without the `cgroup2` support, so it is not surprising that it fails.

**Comment by evverx** [comment]
For now, to silence the test, I'm applying the following patch locally:
```diff
diff --git a/test/TEST-19-DELEGATE/testsuite.sh b/test/TEST-19-DELEGATE/testsuite.sh
index 3fbf347..34b63d1 100755
--- a/test/TEST-19-DELEGATE/testsuite.sh
+++ b/test/TEST-19-DELEGATE/testsuite.sh
@@ -4,16 +4,20 @@
 set -ex
 set -o pipefail

-systemd-run --wait --unit=test0.service -p "DynamicUser=1" -p "Delegate=" \
-            test -w /sys/fs/cgroup/system.slice/test0.service/ -a \
-                 -w /sys/fs/cgroup/system.slice/test0.service/cgroup.procs -a \
-                 -w /sys/fs/cgroup/system.slice/test0.service/cgroup.subtree_control
+if [[ "$(stat -f --format="%T" /sys/fs/cgroup)" = "tmpfs" ]]; then
+       printf "tmpfs is mounted on /sys/fs/cgroup, skipping.\n"
+else
+       systemd-run --wait --unit=test0.service -p "DynamicUser=1" -p "Delegate=" \
+                   test -w /sys/fs/cgroup/system.slice/test0.service/ -a \
+                        -w /sys/fs/cgroup/system.slice/test0.service/cgroup.procs -a \
+                        -w /sys/fs/cgroup/system.slice/test0.service/cgroup.subtree_control

-systemd-run --wait --unit=test1.service -p "DynamicUser=1" -p "Delegate=memory pids" \
-            grep memory /sys/fs/cgroup/system.slice/test1.service/cgroup.controllers
+       systemd-run --wait --unit=test1.service -p "DynamicUser=1" -p "Delegate=memory pids" \
+                   grep memory /sys/fs/cgroup/system.slice/test1.service/cgroup.controllers

-systemd-run --wait --unit=test2.service -p "DynamicUser=1" -p "Delegate=memory pids" \
-            grep pids /sys/fs/cgroup/system.slice/test2.service/cgroup.controllers
+       systemd-run --wait --unit=test2.service -p "DynamicUser=1" -p "Delegate=memory pids" \
+                   grep pids /sys/fs/cgroup/system.slice/test2.service/cgroup.controllers
+fi

 echo OK > /testok

diff --git a/test/test-functions b/test/test-functions
index a2f8272..40243bc 100644
--- a/test/test-functions
+++ b/test/test-functions
@@ -21,7 +21,7 @@ if ! ROOTLIBDIR=$(pkg-config --variable=systemdutildir systemd); then
     ROOTLIBDIR=/usr/lib/systemd
 fi

-BASICTOOLS="test sh bash setsid loadkeys setfont login sulogin gzip sleep echo mount umount cryptsetup date dmsetup modprobe sed cmp tee rm true false"
+BASICTOOLS="stat test sh bash setsid loadkeys setfont login sulogin gzip sleep echo mount umount cryptsetup date dmsetup modprobe sed cmp tee rm true false"
 DEBUGTOOLS="df free ls stty cat ps ln ip route dmesg dhclient mkdir cp ping dhclient strace less grep id tty touch du sort hostname find"

 STATEDIR="${BUILD_DIR:-.}/test/$(basename $(dirname $(realpath $0)))"
```
but it would be great if someone else who understands what this test is intended to test better than I do could fix it properly.

**Comment by poettering** [comment]
Proposed fix waiting in #7445. PTAL!
